File: ChatGPT/portscan.c
package main

import (
	"fmt"
	"net"
	"time"
)

func main() {
	// Set the IP address and port range to scan
	ip := "TARGET_IP_ADDRESS"
	startPort := 1
	endPort := 65535

	// Loop through the port range and try to establish a connection to each port
	for port := startPort; port <= endPort; port++ {
		// Create a connection string in the format "IP:PORT"
		connString := fmt.Sprintf("%s:%d", ip, port)

		// Try to establish a connection to the current port
		conn, err := net.DialTimeout("tcp", connString, time.Millisecond*100)
		if err == nil {
			// The connection was successful, so the port is open
			fmt.Printf("Port %d is open\n", port)
			conn.Close()
		}
	}
}

File: ChatGPT/portscan.py
import socket

def port_scanner(ip, ports):
    # Create a socket object
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    # Set the timeout to 1 second
    s.settimeout(1)

    # Loop through the list of ports
    for port in ports:
        # Try to connect to the target IP and port
        try:
            con = s.connect((ip, port))
            print("Port", port, "is open")
            # Close the connection
            con.close()
        except:
            # If there is an error, the port is likely closed
            continue

# Test the port scanner with the target IP and a list of ports
ip = "TARGET_IP_ADDRESS"
ports = [22, 80, 443, 3389]
port_scanner(ip, ports)
